algoliasearch/insights_request.go

```diff
@@ -0,0 +1,36 @@
+package algoliasearch
+
+import "time"
+
+type insightsRequest struct {
+	EventType string   `json:"eventType"`
+	EventName string   `json:"eventName"`
+	Index     string   `json:"index"`
+	UserToken string   `json:"userToken"`
+	Timestamp int64    `json:"timestamp"`
+	QueryID   string   `json:"queryID"`
+	ObjectIDs []string `json:"objectIDs"`
+	Filters   []string `json:"filters"`
+	Positions []int    `json:"positions"`
+}
+
+func newInsightsRequest(userToken, eventName, indexName, eventType string) insightsRequest {
+	return insightsRequest{
+		EventType: eventType,
+		EventName: eventName,
+		Timestamp: time.Now().Unix() * 1000,
```

Comment:
You could remove this as the API will set it to `now()` if not defined.

Reply:
Indeed, thank you 👍 

I've also set `UserToken` field to `userToken` (param) that was missing :/